### ISSUE TITLE
Tab-completion is not working as expected since version 7.2: no or incomplete suggestions, lag

### ISSUE THREAD

**Issue by normanius** [ipython/ipython#11908]
I describe here a problem related to the tab-completion feature. For recent versions of IPython, tab-completion does not work the way it used to work for version 7.1 (and possibly earlier).

Besides not working as expected, tab-completion (in the cases where it still works) introduces a noticeable lag, which was not present for IPython 7.1. 

The underlying problem is certainly a duplicate of #11530, but that issue refers to paths. I assume the problem is more generic.

The problem(s) appear starting from IPython 7.2, and are still present for IPython 7.8.

My system:
MacOS 10.14.5
iTerm2 / zsh
python 3.6.7
ipython >=7.2.0
(pandas 0.24.2)

Here is how I can reproduce the problem:

```
import pandas as pd
df = pd.DataFrame(columns=['col_a', 'col_b'], data=np.random.random((3,2)))

# shows nothing
df.<TAB>

# Shows some methods and properties starting with 'co'. See screenshots 
# [1]-[3]. The difference with version 7.1 (screenshot [3]) are: 
#    (A) The list of suggested attributes are missing some entries. 
#    (B) Methods can be distinguished from properties (notice the brackets)
# Well, this new feature (B) was present for IPython 7.2-7.7, but has been 
# removed again for 7.8. Compare screenshot [1] with screenshot [2].
df.co<TAB>

# going one level deeper into the object:

# shows nothing
df.columns.<TAB>
# again shows nothing
df.columns.get<TAB>
```

If I downgrade to version 7.1, ipython, everything works as expected:

```
import pandas as pd
df = pd.DataFrame(columns=['col_a', 'col_b'], data=[[1,2],[3,4]])

# Shows all methods and attributes
df.<TAB>

# Shows all methods and attributes starting with 'co'
# See screenshot [3].
df.co<TAB>

# Tab completion also works when going one level deeper:
# See screenshot [4]
df.columns.<TAB>
# Shows all methods / attributes
df.columns.get<TAB>
```

Screenshot [1]: (applies to IPython version 7.8)
![image](https://user-images.githubusercontent.com/11406002/66579503-35f1f700-eb7d-11e9-996d-519e17a8b187.png)

Screenshot [2]: (applies to IPython version 7.2, notice the brackets)
![image](https://user-images.githubusercontent.com/11406002/66577885-697f5200-eb7a-11e9-98a6-96e40a748741.png)

Screenshot [3]: (applies to IPython version 7.1)
![image](https://user-images.githubusercontent.com/11406002/66578387-4b662180-eb7b-11e9-9d1b-413d7848f5cd.png)

Screenshot [4]: (applies only to IPython version 7.1)
![image](https://user-images.githubusercontent.com/11406002/66578246-09d57680-eb7b-11e9-8342-3e829d4c3873.png)

**Comment by normanius** [comment]
Well, a small update: in version 7.8, the brackets are still shown, but only after a second attempt using tab.... (That's why they are visible in screenshot [3]). This problem is new in version 7.8 - I didn't observe it for 7.2-7.7... 

In any case, something looks messed up. And it's not just my bug-report :)

**Comment by augustogoulart** [comment]
@meeseeksdev tag help wanted

**Comment by augustogoulart** [comment]
@normanius thanks for reporting this. I tested the tab-completion on 7.9, and it seems to be working as expected, however, I did notice a considerable lag on 7.9 that’s not present on 7.1.

:movie_camera: Video recording for 7.9:
https://recordit.co/aKoZsujIHx

:movie_camera: Video recording for 7.1:
https://recordit.co/f42GQ82Jb2

With that said, I believe this may be a performance issue and not a broken feature.

**Comment by augustogoulart** [comment]
@meeseeksdev untag help wanted

**Comment by augustogoulart** [comment]
@normanius I created https://github.com/ipython/ipython/issues/11942 to address the lag issue.
Please let us know if you want to add something else to this. Thanks!

**Comment by augustogoulart** [comment]
@meeseeksdev close

**Comment by normanius** [comment]
@augustogoulart Thanks for looking into this. I don't think this issue should be closed, because in my view, tab completion is still broken.

I upgraded to ipython 7.9.0 created a new shell session (I'm using zsh, but this should not matter, I guess) and tried the following sequence of commands.

```
# 
import pandas as pd
df = pd.DataFrame(columns=['col_a', 'col_b'], data=[[1,2],[3,4]])

# [NOT OKAY] Doesn't show anything (see first screenshot below)
df.<TAB>

# [LOOKS OKAY] Shows all methods and attributes starting with 'co'
df.co<TAB>

# Tab completion when going one level deeper into the object:
# [NOT OKAY] Doesn't show anything, see second screenshot below
df.columns.<TAB>
# [NOT OKAY] Doesn't show anything
df.columns.get<TAB>
```

![image](https://user-images.githubusercontent.com/11406002/68853511-6ed03f00-06da-11ea-8d30-2134e53a5409.png)

![image](https://user-images.githubusercontent.com/11406002/68853659-cc648b80-06da-11ea-8b0e-7cda25620833.png)

In summary: The tab-completion functionality of 7.1. is still superior to 7.9.

Let me know how I can contribute to this issue.

**Comment by normanius** [comment]
Also note that for some modules, I cannot see any content:

```
import pandas as pd
import numpy as np

# [OKAYISH] With some lag (see other issue), a list of suggestions is shown.
pd.<TAB>
# [NOT OKAY] No suggestions are shown
np.<TAB>
```
